backend/test/experiment_api_test.go

```diff
@@ -0,0 +1,161 @@
+package test
+
+import (
+	"context"
+	"testing"
+
+	"time"
+
+	"github.com/golang/glog"
+	api "github.com/kubeflow/pipelines/backend/api/go_client"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc"
+)
+
+type ExperimentApiTest struct {
+	suite.Suite
+	namespace        string
+	conn             *grpc.ClientConn
+	experimentClient api.ExperimentServiceClient
+}
+
+// Check the namespace have ML job installed and ready
+func (s *ExperimentApiTest) SetupTest() {
+	err := waitForReady(*namespace, *initializeTimeout)
+	if err != nil {
+		glog.Exitf("Failed to initialize test. Error: %s", err.Error())
+	}
+	s.namespace = *namespace
+	s.conn, err = getRpcConnection(s.namespace)
+	if err != nil {
+		glog.Exitf("Failed to get RPC connection. Error: %s", err.Error())
+	}
+	s.experimentClient = api.NewExperimentServiceClient(s.conn)
```

Comment:
That library has a lot of duplicated code that we should clean them up before switching the code here to use it. (Otherwise I would have to add more duplication there) Can we do it in a separate PR? The API server is now running naked. I want to get test in place ASAP.

Reply:
The goal of the wrapper is to be used by anything that needs to make the call in GO. Let's please switch to using it so that we don't duplicate work. For instance, be adding retries/timeout in the wrapper, any user of the wrapper immediately benefits from it.

Reply:
The goal of the wrapper is to be used by anything that needs to make the call in GO. Let's please switch to using it so that we don't duplicate work. For instance, be adding retries/timeout in the wrapper, any user of the wrapper immediately benefits from it.

If the wrapper does things that are problematic for the API e2e test, let's modify the wrapper accordingly.

Reply:
I can use it once it's ready. Right now it's missing quite a few clients and it's using HTTP which is not what being used by tests. Let;s add a todo and prioritize that work. The test itself is ready and really high priority.